File: src/quick-bite-payment/src/main/java/com/quickbite/payment/application/dto/PaymentFailedEto.java
package com.quickbite.payment.application.dto;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Event Transfer Object published when a Payment process fails.
 * Maps to PaymentFailedEto in Order Service.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PaymentFailedEto {
    private UUID eventId;
    private UUID orderId;
    private UUID paymentId;
    private String reason;
    private UUID customerId;
    private BigDecimal amount;
    private String eventType;
    private UUID correlationId;
    private LocalDateTime occurredAt;
}

File: src/quick-bite-order/QuickBite.Order/src/QuickBite.Order.Domain.Shared/Event/PaymentFailedEto.cs
using System;
using Volo.Abp.EventBus;

namespace QuickBite.Order.Domain.Shared.Event;

/// <summary>
/// Event emitted by Payment Service when payment authorization fails.
/// Published to "fulfillment-events" topic.
/// </summary>
[EventName("payment.failed")]
public class PaymentFailedEto
{
    public Guid EventId { get; set; }

    public Guid OrderId { get; set; }

    public string Reason { get; set; }

    public Guid CorrelationId { get; set; }

    public DateTime OccurredAt { get; set; }
}
